### PR TITLE
Clarification on the payload doc string

### PR DESCRIPTION
**Pull request by krishnakalyan3**
## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃
cc: @tchaton , @awaelchli

### CODE REVIEW

**Review by alecmerdler — APPROVED** [review]
Thanks for the clarification!

File: src/lightning_app/storage/payload.py [hunk 1]
```diff
@@ -239,7 +239,9 @@ def _handle_get_request(work: "lightning_app.LightningWork", request: GetRequest
 class Payload(BasePayload):
 
     """The Payload object enables to transfer python objects from one work to another in a similar fashion as
-    :class:`~lightning_app.storage.path.Path`."""
+    :class:`~lightning_app.storage.path.Path`.
+    Payload is similar to Drive, Payload object is a pickled file that is unpickled into another other work.
```
Comment (otaj) [inline src/lightning_app/storage/payload.py:243]:
```suggestion
    Payload is similar to Drive, Payload object is a pickled file that is unpickled into another work.
```

**Review by otaj — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: src/lightning_app/storage/payload.py [hunk 2]
```diff
@@ -239,7 +239,9 @@ def _handle_get_request(work: "lightning_app.LightningWork", request: GetRequest
 class Payload(BasePayload):
 
     """The Payload object enables to transfer python objects from one work to another in a similar fashion as
-    :class:`~lightning_app.storage.path.Path`."""
+    :class:`~lightning_app.storage.path.Path`.
```
Comment (lantiga) [inline src/lightning_app/storage/payload.py:242]:
```suggestion
    :class:`~lightning_app.storage.payload.Payload`.
```

File: src/lightning_app/storage/payload.py [hunk 3]
```diff
@@ -239,7 +239,9 @@ def _handle_get_request(work: "lightning_app.LightningWork", request: GetRequest
...
+    Payload is similar to Drive, Payload object is a pickled file that is unpickled into another work.
```
Comment (lantiga) [inline src/lightning_app/storage/payload.py:243]:
```suggestion
    `Payload` is used to hold large objects as part of the app state. In this case the state variable will only hold a reference to the object, while the object itself will be serialized and transferred outside the state. Currently `Payload` serializes using `pickle` and transfers through `s3` (similarly to `Drive`), but `BasePayload` can be subclassed to support different serialization and transfer mechanisms.
```

**Review by lantiga — CHANGES_REQUESTED** [review]

**Pull request closed without merging**